test/spec_server.rb

```diff
@@ -141,7 +141,8 @@ def with_stderr
   it "check pid file presence and running process" do
     pidfile = Tempfile.open('pidfile') { |f| f.write($$); break f }.path
     server = Rack::Server.new(:pid => pidfile)
-    server.send(:pidfile_process_status).must_equal :running
+    server.send(:pidfile_process_status).must_equal :this_process
+    `ruby -rrack/server -e 'print Rack::Server.new(:pid => "#{pidfile}").send(:pidfile_process_status).inspect'`.must_equal ":running"
```

Comment:
We should use the running ruby, not whatever happens to be on the path.

And similarly, better if we can avoid relying on what library [versions] it can find.

Along these lines, we could instead run a `-e` that'll ~ just sit, then do the pid check in the current process.

(I'd personally be fine with a test that just skipped where fork was unavailable, but as long as the complexity doesn't grow much, this is good too.)